include/stl2/view/drop.hpp

```diff
@@ -75,7 +75,13 @@ STL2_OPEN_NAMESPACE {
 			{
 				auto compute_begin = [&x]{
 					STL2_EXPECT(x.count_ >= 0);
-					return __stl2::next(__stl2::begin(x.base_), x.count_, __stl2::end(x.base_));
+					if constexpr (SizedRange<__maybe_const<is_const_v<X>, R>>) {
+						D const dist = __stl2::distance(x.base_);
+						return __stl2::next(__stl2::begin(x.base_), dist < x.count_ ? dist : x.count_);
```

Comment:
Final nitpick and then I swear I'll stop: advancing `begin(rng)` by `distance(rng)` is unnecessarily O(N) in if the range is `Bounded` but not `RandomAccess`; we could use `end(rng)` directly. How about:
```c++
D n = __stl2::distance(x.base_);
if (n > x.count_) {
    n = x.count_;
} else if constexpr (!RandomAccessRange<X> && BoundedRange<X>) {
    return __stl2::end(x.base_);
}
return __stl2::next(__stl2::begin(x.base_), n);
```

...and now I'm thinking that it would be an improvement to decrement `end` if `x.count_ > distance(x.base_) / 2` and the range is `Bounded` and `Bidirectional`:

```c++
D const size = __stl2::distance(x.base_);
D n = x.count_;
constexpr bool BoundedNonRandom = BoundedRange<X> && !RandomAccessRange<X>;
if (n >= size) {
    if constexpr (BoundedNonRandom && !BidirectionalRange<X>) {
        return __stl2::end(x.base_);
    }
    n = size;
}
if constexpr (BoundedNonRandom && BidirectionalRange<X>) {
    if (n > size / 2) {
        return __stl2::prev(__stl2::end(x,base_), size - n));
    }
}
return __stl2::next(__stl2::begin(x.base_), n);
```
There's a *lot* of subtlety here: we ought to wrap up this entire lambda and propose it as `nth_iterator(range, n)` (or `index`, or `at`, or ...)

Reply:
Where `BoundedRange` is `CommonRange`?

Reply:
> There's a lot of subtlety here: Should we wrap up this entire lambda and propose it as `nth_iterator(range, n)` (or `index`, or `at`, or ...)?

IMO iff `rng | drop(n)` isn't quite enough to always say "start from `n`". Otherwise I guess it could be exposition-only for `drop` and friends?

Reply:
Cursed rename - I'm still not used to it. I suspect there will be other use cases for `nth_iterator`; we can propose it if and when they appear and validate my suspicions.